plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,356 @@
+        elif counter == 13:
+          new_event.ws_man_stack_version = value
+        # remoting protocol version
+        elif counter == 14:
+          new_event.remoting_protocol_version = value
+        # serialization version
+        elif counter == 15:
+          new_event.serialization_version = value
+        counter += 1
+      # update base event for future use
+      self.base_event = copy.deepcopy(new_event)
+    # transcript with timestamp header
+    elif len(metadata_lines) == 1:
+      if self.base_event is None:
+        parser_mediator.ProduceExtractionWarning(
+            'initial transcript seems to have missing headers'
+            ' - can not produce event')
+        return False
+      # since we don't have the full headers,
+      # we just copy the headers from the last full event...
+      new_event = copy.deepcopy(self.base_event)
+      if ': ' in metadata_lines[0]:
+        value = metadata_lines[0].split(': ')[1]
+      else:
+        value = None
+      # ... and overwrite the timestamp
+      start_time = self._GetTimestampFromString(value)
+      if start_time is None:
+        parser_mediator.ProduceExtractionWarning(
+            'timestamp \"{0!s}\" seems invalid - skipping malformed '
+            'transcript'.format(value))
+        return False
+      start_time.is_local_time = True
+      event_time = time_events.DateTimeValuesEvent(start_time,
+        definitions.TIME_DESCRIPTION_START)
+
+    if transcript_text == '':
+      parser_mediator.ProduceExtractionWarning(
+          'skipping transcript {0!s} - since transcript text could '
+          'not be set'.format(key_structure))
+      return False
+    new_event.command = transcript_text
+    # create event if at least the most relevant fields are set
+    if (event_time is not None and new_event.command is not None
+        and new_event.command != ''):
+      parser_mediator.ProduceEventWithEventData(event_time, new_event)
+    else:
+      parser_mediator.ProduceExtractionWarning(
+          'skipping transcript {0!s} - relevant event values could '
+          'not be set'.format(key_structure))
+      return False
+    return True
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parse the record and decide when a full transcript has been seen.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    # skip the very first (separator) line since it does not contain any
+    # relevant info and will break our logic
+    if self.is_first_line:
+      self.is_first_line = False
+      return
+    # the first line of a transcript will always be a metadata line, however
```

Comment:
what is the first line of a transcript in your perspective? is that the line after the second `**********************` ?

Reply:
or is that e.g. `Windows PowerShell transcript start` ?

Reply:
I'll assume the latter

Reply:
I'd also vote for the latter.